certbot/log.py

```diff
@@ -299,7 +300,23 @@ def except_hook(exc_type, exc_value, trace, debug, log_path):
             _, _, exc_str = str(exc_value).partition(':: ')
             print(exc_str, file=sys.stderr)
         else:
-            traceback.print_exception(exc_type, exc_value, None)
+            """ Regex for parsing error messages"""
+            """ Regular expression groups"""
+            """ Match Group #1: Attempted connection host"""
+            """ Match Group #2: urllib3 Error Number"""
+            """ Match Group #3: urllib3 Error String"""
+            """ Example:"""
+            """ Match Group #1: acme-v01.api.letsencrypt.org"""
+            """ Match Group #2: [Errno 65]"""
+            """ Match Group #3: No route to host"""
+            err_re = re.search(r"host='(\S*)'.*(\[Errno \d+\]\ )(.*)(',\)\))", str(exc_value))
+            if err_re:
+                err_msg = "Can't connect to "+err_re.group(1)+\
```

Comment:
It's unusual to separate errors onto multiple lines like that. Also, we can make things clearer by combining the host, path, and scheme. E.g.:

> Requesting https://acme-v01.api.letsencrypt.org/directory: [Errno -2] Name or service not known